x/valset-pref/msg_server_test.go

```diff
@@ -105,6 +107,59 @@ func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
 			},
 			expectPass: false,
 		},
+		{
+			name:      "user doesnot have valset, but has existing delegation",
+			delegator: sdk.AccAddress([]byte("addr2---------------")),
+			preferences: []types.ValidatorPreference{
+				{
+					ValOperAddress: valAddrs[3],
+					Weight:         sdk.NewDecWithPrec(5, 1),
+				},
+				{
+					ValOperAddress: valAddrs[4],
+					Weight:         sdk.NewDecWithPrec(5, 1),
+				},
+			},
+			coin:                   sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(10_000_000)),
+			setExistingDelegations: true,
+			expectPass:             true, //SetValidatorSetPreference sets the existing delegations as valset and ignores test.preferences
+		},
+		{
+			name:      "user has valset, but doesnot have existing delegation",
+			delegator: sdk.AccAddress([]byte("addr1---------------")),
+			preferences: []types.ValidatorPreference{
+				{
+					ValOperAddress: valAddrs[3],
+					Weight:         sdk.NewDecWithPrec(3, 1),
+				},
+				{
+					ValOperAddress: valAddrs[4],
+					Weight:         sdk.NewDecWithPrec(7, 1),
+				},
+			},
+			expectPass: true, // SetValidatorSetPreference modifies the existing delegations
+		},
+		{
+			name:      "user has existing valset and existing delegation",
+			delegator: sdk.AccAddress([]byte("addr1---------------")),
+			preferences: []types.ValidatorPreference{
+				{
+					ValOperAddress: valAddrs[3],
+					Weight:         sdk.NewDecWithPrec(3, 1),
+				},
+				{
+					ValOperAddress: valAddrs[4],
+					Weight:         sdk.NewDecWithPrec(3, 1),
+				},
+				{
+					ValOperAddress: valAddrs[5],
```

Comment:
just curious, why do we use new validators( valAddr[3], [4] [5] and so on instead of 0, 1, 2)?

Reply:
So in this case we'll update the valset to preferences. For addr1, existing valsets are Val3 (0.3) , Val4 (0.7)  from [this test case ](https://github.com/osmosis-labs/osmosis/pull/3857/files/e0878a4517045f842f5afec8c92033adde0d70a4#diff-e8d94c443b26e1cda354d150b30ce5542344099e0bbbb8e84e507d9df791daf7R128-R140)

Now we want to update to Val3(0.3), Val4(0.3), Val5(0.4). We can use Val0, Val1, Val2 but the functionality will be the same